homeassistant/components/sonos/speaker.py

```diff
@@ -553,25 +554,29 @@ def speaker_activity(self, source):
 
     async def async_check_activity(self, now: datetime.datetime) -> None:
         """Validate availability of the speaker based on recent activity."""
+        if not self.available:
+            return
         if time.monotonic() - self._last_activity < AVAILABILITY_TIMEOUT:
             return
 
         try:
-            _ = await self.hass.async_add_executor_job(getattr, self.soco, "volume")
-        except (OSError, SoCoException):
-            pass
+            # Make a short-timeout call as a final check
+            # before marking this speaker as unavailable
+            await self.hass.async_add_executor_job(
+                partial(
+                    self.soco.renderingControl.GetVolume,
+                    [("InstanceID", 0), ("Channel", "Master")],
+                    timeout=1,
+                )
+            )
+        except OSError:
+            _LOGGER.warning(
```

Comment:
This should be info or debug?

Reply:
It's a problem condition that wasn't automatically recovered from when connectivity failed, so I'd think either warning or error would be most appropriate.

Reply:
If you say so.